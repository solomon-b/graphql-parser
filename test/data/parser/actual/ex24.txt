[ DefinitionExecutable
    ( ExecutableDefinitionOperation
        ( OperationDefinition
            { _odType = OperationTypeQuery
            , _odName = Just "inlineFragmentNoType"
            , _odVariables = Just
                ( VariableDefinition
                    { _vdName = "expandedInfo"
                    , _vdType = NamedType "Boolean"
                    , _vdDefaultValue = Nothing
                    , _vdDirectives = Nothing
                    } :| []
                )
            , _odDirectives = Nothing
            , _odSelectionSet = SelectionField
                ( Field
                    { _fAlias = Nothing
                    , _fName = "user"
                    , _fArguments = fromList
                        [
                            ( "handle"
                            , VString "zuck"
                            )
                        ]
                    , _fDirectives = Nothing
                    , _fSelectionSet = Just
                        ( SelectionField
                            ( Field
                                { _fAlias = Nothing
                                , _fName = "id"
                                , _fArguments = fromList []
                                , _fDirectives = Nothing
                                , _fSelectionSet = Nothing
                                }
                            ) :|
                            [ SelectionField
                                ( Field
                                    { _fAlias = Nothing
                                    , _fName = "name"
                                    , _fArguments = fromList []
                                    , _fDirectives = Nothing
                                    , _fSelectionSet = Nothing
                                    }
                                )
                            , SelectionInlineFragment
                                ( InlineFragment
                                    { _ifTypeCondition = Nothing
                                    , _ifDirectives = Just
                                        ( Directive
                                            { dName = "include"
                                            , dArguments = Just
                                                ( fromList
                                                    [
                                                        ( "if"
                                                        , VVar "expandedInfo"
                                                        )
                                                    ]
                                                )
                                            } :| []
                                        )
                                    , _ifSelectionSet = SelectionField
                                        ( Field
                                            { _fAlias = Nothing
                                            , _fName = "firstName"
                                            , _fArguments = fromList []
                                            , _fDirectives = Nothing
                                            , _fSelectionSet = Nothing
                                            }
                                        ) :|
                                        [ SelectionField
                                            ( Field
                                                { _fAlias = Nothing
                                                , _fName = "lastName"
                                                , _fArguments = fromList []
                                                , _fDirectives = Nothing
                                                , _fSelectionSet = Nothing
                                                }
                                            )
                                        , SelectionField
                                            ( Field
                                                { _fAlias = Nothing
                                                , _fName = "birthday"
                                                , _fArguments = fromList []
                                                , _fDirectives = Nothing
                                                , _fSelectionSet = Nothing
                                                }
                                            )
                                        ]
                                    }
                                )
                            ]
                        )
                    }
                ) :| []
            }
        )
    )
]