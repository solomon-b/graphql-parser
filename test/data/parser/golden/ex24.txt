[ DefinitionExecutable
    ( ExecutableDefinitionOperation
        ( OperationDefinition
            { _odSpan = Span
                { _start = AlexSourcePos
                    { _line = 0
                    , _col = 1
                    }
                , _end = AlexSourcePos
                    { _line = 10
                    , _col = 2
                    }
                }
            , _odType = OperationTypeQuery
            , _odName = Just "inlineFragmentNoType"
            , _odVariables = Just
                ( VariableDefinition
                    { _vdSpan = Span
                        { _start = AlexSourcePos
                            { _line = 0
                            , _col = 28
                            }
                        , _end = AlexSourcePos
                            { _line = 0
                            , _col = 50
                            }
                        }
                    , _vdName = "$expandedInfo"
                    , _vdType = NamedType
                        ( Span
                            { _start = AlexSourcePos
                                { _line = 0
                                , _col = 43
                                }
                            , _end = AlexSourcePos
                                { _line = 0
                                , _col = 50
                                }
                            }
                        ) "Boolean"
                    , _vdDefaultValue = Nothing
                    , _vdDirectives = Nothing
                    } :| []
                )
            , _odDirectives = Nothing
            , _odSelectionSet = SelectionField
                ( Field
                    { _fSpan = Span
                        { _start = AlexSourcePos
                            { _line = 1
                            , _col = 3
                            }
                        , _end = AlexSourcePos
                            { _line = 9
                            , _col = 4
                            }
                        }
                    , _fAlias = Nothing
                    , _fName = "user"
                    , _fArguments = Just
                        ( Arguments
                            { argSpan = Span
                                { _start = AlexSourcePos
                                    { _line = 1
                                    , _col = 8
                                    }
                                , _end = AlexSourcePos
                                    { _line = 1
                                    , _col = 22
                                    }
                                }
                            , argArguments = fromList
                                [
                                    ( "handle"
                                    , VString
                                        ( Span
                                            { _start = AlexSourcePos
                                                { _line = 1
                                                , _col = 16
                                                }
                                            , _end = AlexSourcePos
                                                { _line = 1
                                                , _col = 22
                                                }
                                            }
                                        ) "zuck"
                                    )
                                ]
                            }
                        )
                    , _fDirectives = Nothing
                    , _fSelectionSet = Just
                        ( SelectionField
                            ( Field
                                { _fSpan = Span
                                    { _start = AlexSourcePos
                                        { _line = 2
                                        , _col = 5
                                        }
                                    , _end = AlexSourcePos
                                        { _line = 2
                                        , _col = 7
                                        }
                                    }
                                , _fAlias = Nothing
                                , _fName = "id"
                                , _fArguments = Nothing
                                , _fDirectives = Nothing
                                , _fSelectionSet = Nothing
                                }
                            ) :|
                            [ SelectionField
                                ( Field
                                    { _fSpan = Span
                                        { _start = AlexSourcePos
                                            { _line = 3
                                            , _col = 5
                                            }
                                        , _end = AlexSourcePos
                                            { _line = 3
                                            , _col = 9
                                            }
                                        }
                                    , _fAlias = Nothing
                                    , _fName = "name"
                                    , _fArguments = Nothing
                                    , _fDirectives = Nothing
                                    , _fSelectionSet = Nothing
                                    }
                                )
                            , SelectionInlineFragment
                                ( InlineFragment
                                    { _ifSpan = Span
                                        { _start = AlexSourcePos
                                            { _line = 4
                                            , _col = 5
                                            }
                                        , _end = AlexSourcePos
                                            { _line = 8
                                            , _col = 6
                                            }
                                        }
                                    , _ifTypeCondition = Nothing
                                    , _ifDirectives = Just
                                        ( Directive
                                            { _dSpan = Span
                                                { _start = AlexSourcePos
                                                    { _line = 4
                                                    , _col = 9
                                                    }
                                                , _end = AlexSourcePos
                                                    { _line = 4
                                                    , _col = 35
                                                    }
                                                }
                                            , _dName = "include"
                                            , _dArguments = Just
                                                ( Arguments
                                                    { argSpan = Span
                                                        { _start = AlexSourcePos
                                                            { _line = 4
                                                            , _col = 18
                                                            }
                                                        , _end = AlexSourcePos
                                                            { _line = 4
                                                            , _col = 35
                                                            }
                                                        }
                                                    , argArguments = fromList
                                                        [
                                                            ( "if"
                                                            , VVar
                                                                ( Span
                                                                    { _start = AlexSourcePos
                                                                        { _line = 4
                                                                        , _col = 22
                                                                        }
                                                                    , _end = AlexSourcePos
                                                                        { _line = 4
                                                                        , _col = 35
                                                                        }
                                                                    }
                                                                ) "$expandedInfo"
                                                            )
                                                        ]
                                                    }
                                                )
                                            } :| []
                                        )
                                    , _ifSelectionSet = SelectionField
                                        ( Field
                                            { _fSpan = Span
                                                { _start = AlexSourcePos
                                                    { _line = 5
                                                    , _col = 7
                                                    }
                                                , _end = AlexSourcePos
                                                    { _line = 5
                                                    , _col = 16
                                                    }
                                                }
                                            , _fAlias = Nothing
                                            , _fName = "firstName"
                                            , _fArguments = Nothing
                                            , _fDirectives = Nothing
                                            , _fSelectionSet = Nothing
                                            }
                                        ) :|
                                        [ SelectionField
                                            ( Field
                                                { _fSpan = Span
                                                    { _start = AlexSourcePos
                                                        { _line = 6
                                                        , _col = 7
                                                        }
                                                    , _end = AlexSourcePos
                                                        { _line = 6
                                                        , _col = 15
                                                        }
                                                    }
                                                , _fAlias = Nothing
                                                , _fName = "lastName"
                                                , _fArguments = Nothing
                                                , _fDirectives = Nothing
                                                , _fSelectionSet = Nothing
                                                }
                                            )
                                        , SelectionField
                                            ( Field
                                                { _fSpan = Span
                                                    { _start = AlexSourcePos
                                                        { _line = 7
                                                        , _col = 7
                                                        }
                                                    , _end = AlexSourcePos
                                                        { _line = 7
                                                        , _col = 15
                                                        }
                                                    }
                                                , _fAlias = Nothing
                                                , _fName = "birthday"
                                                , _fArguments = Nothing
                                                , _fDirectives = Nothing
                                                , _fSelectionSet = Nothing
                                                }
                                            )
                                        ]
                                    }
                                )
                            ]
                        )
                    }
                ) :| []
            }
        )
    )
]